test/e2e/monitoring/alerts_test.go

```diff
@@ -139,5 +147,82 @@ var _ = Context("Prometheus Alerts", func() {
 				ScaleDeployment(components.Name, components.Namespace, 1)
 			})
 		})
+
+		Context("and there are duplicate MACs", func() {
+			var virtClient kubecli.KubevirtClient
+			var err error
+
+			BeforeEach(func() {
+				By("starting virtClient")
+				virtClient, err = kubecli.GetKubevirtClientFromFlags("", os.Getenv("KUBECONFIG"))
+				Expect(err).ToNot(HaveOccurred())
+
+				By("creating test namespace")
+				namespace := &k8sv1.Namespace{ObjectMeta: metav1.ObjectMeta{
+					Name: kvtutil.NamespaceTestDefault,
+					Labels: map[string]string{
+						"mutatevirtualmachines.kubemacpool.io": "ignore",
+					},
+				}}
+				_, err = virtClient.CoreV1().Namespaces().Create(context.Background(), namespace, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				By("deleting test namespace")
+				err = virtClient.CoreV1().Namespaces().Delete(context.Background(), kvtutil.NamespaceTestDefault, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("should issue KubeMacPoolDuplicateMacsFound alert", func() {
+				By("creating 2 VMs with a duplicate MAC")
+				err = createVirtualMachine(virtClient)
+				Expect(err).ToNot(HaveOccurred())
+				err = createVirtualMachine(virtClient)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("cleaning namespace labels")
+				err = cleanNamespaceLabels(virtClient, kvtutil.NamespaceTestDefault)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("restaring kubemacpool pods")
```

Comment:
IMO as the test only checks the alert, all the preparations (i.e. creating the VMs, removing the labels, restarting kubemacpool) should be in BeforeEach section.
keeping the preparations out of the `It` scope helps the reader understand what exactly is tested.

Reply:
done